Milliehartnt123 update architectural overview

milliehartnt123 opened the pull request:
## Hey, I just made a Pull Request!

Rewrote the Architecture Overview. Redrew several of the graphics.

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

aramissennyeydd commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,224 @@ Backstage is organized into three main components, each catering to different gr
-Third-party backed plugins are similar to service backed plugins. The main
-difference is that the service which backs the plugin is hosted outside of the
-ecosystem of the company hosting Backstage.
+Third-party backed plugins are similar to service backed plugins. The main difference is that the service which backs the plugin is hosted outside of the ecosystem of the company hosting Backstage.
```
```suggestion
Third-party backend plugins are similar to service backend plugins. The main difference is that the service which backs the plugin is hosted outside of the ecosystem of the company hosting Backstage.
```

aramissennyeydd commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,224 @@ Backstage is organized into three main components, each catering to different gr
-which are cache stores better suited for production deployment. The right cache
-store for your Backstage instance will depend on your own run-time constraints
-and those required of the plugins you're running.
+At this time of writing, Backstage can be configured to use one of five cache stores: memory, which is mainly used for local testing, memcache, redis, valkey or infinispan, which are cache stores better suited for production deployment. The right cache store for your Backstage instance will depend on your own run-time constraints and those required of the plugins you're running.
```
```suggestion
At this time of writing, Backstage can be configured to use one of five cache stores: `memory`, `memcache`, `redis`, `valkey` or `infinispan`. `memory` is primarily used for local development, for production deployments, we recommend using one of the other cache stores. The right cache store for your Backstage instance will depend on your own run-time constraints and those required of the plugins you're running.
```

milliehartnt123 replied:
Made the change. Changed the list of cache stores into a bulleted list.

aramissennyeydd reviewed (commented):
This is awesome!

milliehartnt123 commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,224 @@ Backstage is organized into three main components, each catering to different gr
 
-The architecture of the Tech Radar installed into a Backstage app is very
-simple.
+### Service backed plugins
```
Made the change.

milliehartnt123 commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,224 @@ Backstage is organized into three main components, each catering to different gr
-ecosystem of the company hosting Backstage.
+Third-party backed plugins are similar to service backed plugins. The main difference is that the service which backs the plugin is hosted outside of the ecosystem of the company hosting Backstage.
+
+The CircleCI plugin is an example of a third-party backed plugin. CircleCI is a SaaS service which can be used without any knowledge of Backstage. It has an API which a Backstage plugin consumes to display content.
```
Made the change

milliehartnt123 commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,224 @@ Backstage is organized into three main components, each catering to different gr
-1. The core Backstage UI
-2. The UI plugins and their backing services
-3. Databases
+- The [frontend container](#frontend-building-blocks) includes the core Backstage [UI](#user-interface) which is an [extension](#extensions) that interacts directly with the user to present the information from the integrated core feature plugins, and other plugins added by a user.
```
Changed "Frontend Container" to "Frontend" everywhere in the doc. Changed "Backend Container" to "Backend" everywhere in the doc.

aramissennyeydd commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,231 @@ Backstage is organized into three main components, each catering to different gr
-[https://example.com](https://example.com) from serving resources hosted at
-https://circleci.com.
+> **NOTE:**
+> The following diagram does not show the detailed contents of the frontend and backen, in order to highlight the changes that pertain to the addition of the specified plugin.
```
```suggestion
> The following diagram does not show the detailed contents of the frontend and backend, in order to highlight the changes that pertain to the addition of the specified plugin.
```

aramissennyeydd commented on docs/overview/architecture-overview.md:
```diff
@@ -10,285 +10,231 @@ Backstage is organized into three main components, each catering to different gr
+> The following diagram does not show the detailed contents of the frontend and backend, in order to highlight the changes that pertain to the addition of the specified plugin.
 
-Its architecture looks like this:
+![lighthouse plugin backed to microservice and database](../assets/architecture-overview/simplified-service-based-plugin-architecture.jpeg)
```
```suggestion
![lighthouse plugin backend to microservice and database](../assets/architecture-overview/simplified-service-based-plugin-architecture.jpeg)
```

milliehartnt123 replied:
@aramissennyeydd Made the change.

aramissennyeydd commented:
@milliehartnt123 Looks like prettier isn't happy here - should just need a `yarn prettier:fix` from the root directory. (I can run this as well if you'd like)

milliehartnt123 commented:
@aramissennyeydd  I fixed the prettier error.  I copied the entire file into VS Code where I have prettier running when I save the file. Once I saved the file, I copied the entire file back into GitHub and it passes the prettier check.

I think it is the dashes for the bullets.  I added that section in GitHub since I was incorporating one of the review comments.

Won't do that again.

But thanks for the offer!
